tasks/post-to-cmr/tests/cmr-test.js

```diff
@@ -128,9 +129,61 @@ test.serial('postToCMR skips CMR step if the metadata file uri is missing', asyn
     }]
   }];
 
-  const output = await postToCMR(newPayload);
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+  }
+});
+
+test.serial('postToCMR throws an error if any granule is missing a metadata file', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = {
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.xml`
+    }]
+  };
+  newPayload.input.granules.push(newGranule);
+
+  try {
+    await postToCMR(newPayload);
+  }
+  catch (error) {
+    t.true(error instanceof CMRMetaFileNotFound);
+    t.is(error.message, (`CMR Meta file not found for granule ${newGranule.granuleId}`));
+  }
+});
+
+test.serial('postToCMR continues without metadata file if there is metaCheck flag', async (t) => {
+  const newPayload = t.context.payload;
+  const newGranule = [{
+    granuleId: 'MOD11A1.A2017200.h19v04.006.2017201090555',
+    files: [{
+      filename: `s3://${t.context.bucket}/to/file.xml`
+    }]
+  }];
+  newPayload.input.granules = newGranule;
+  newPayload.config.metaCheck = true;
+  const granuleId = newPayload.input.granules[0].granuleId;
+  const key = `${granuleId}.cmr.xml`;
 
-  t.is(output.granules[0].cmrLink, undefined);
+  sinon.stub(cmrjs.CMR.prototype, 'ingestGranule').callsFake(() => ({
+    result
+  }));
+  try {
+    await aws.promiseS3Upload({
+      Bucket: t.context.bucket,
+      Key: key,
+      Body: fs.createReadStream('tests/data/meta.xml')
+    });
+    const output = await postToCMR(newPayload);
+    t.is(output.granules[0].granuleId, granuleId);
+  }
+  finally {
+    cmrjs.CMR.prototype.ingestGranule.restore();
+  }
```

Comment:
There is one more case that warrants a unit test imo: The passing with config.metaCheck (or skipMetaCheck) if not all granules have metadata, but some do.

Reply:
Added